PHP 8.1 upgrade

mansisheth13 opened the pull request:
- Update dependencies for PHP 8.1 support
- Refactor code so that required parameters are before optional parameters.

Commit:
fix(PHP 8.1 support) Update for PHP 8.1 support

composer.json update dependencies
- nswdpc/silverstripe-pwnage-hinter
- spomky-labs/otphp

ProfileExtension.php
- Change order of parameters so optional parameters are last.

Commit:
rearrange the parameters so that the required parameters are before optional parameters.

Commit:
change version to resolve conflict.

mansisheth13 committed:
Merge branch 'master' into php-8-1-update

JamesDPC commented:
Hi @mansisheth13 I'd updated this on the `updates` branch at  https://github.com/nswdpc/silverstripe-authentication-boilerplate/compare/master...updates, along with some other items.

I'd prefer to maintain the parameter order for the methods and just remove the `false` default to the first arg. It needs to be specified anyway because the 2nd arg is required, doing this will not break anything calling the public method.

Are you able to test out the updates branch to see if this works for you?

All of the other commits are in master and tagged.

Commit:
revert the order of parameters and remove default values.

Commit:
update test method signature.

mansisheth13 commented:
Hi @JamesDPC , 

I've reverted the parameter order for the methods and removed the default values.
I've also updated one of the test's setUp signature to be consistent with SapphireTest.
The module test cases pass. I haven't been able to test much beyond that.

JamesDPC commented:
Hi @mansisheth13 - please see v0.3.1  (https://github.com/nswdpc/silverstripe-authentication-boilerplate/compare/v0.3.0...v0.3.1) as that version includes the updates here, along with a few other changes.

Will close this PR :)
